indexing-service/src/main/java/org/apache/druid/indexing/common/actions/LocalTaskActionClient.java

```diff
@@ -72,14 +77,35 @@ public <RetType> RetType submit(TaskAction<RetType> taskAction)
 
     final long performStartTime = System.currentTimeMillis();
     final RetType result = taskAction.perform(task, toolbox);
-    emitTimerMetric("task/action/run/time", System.currentTimeMillis() - performStartTime);
+    emitTimerMetric("task/action/run/time", taskAction, System.currentTimeMillis() - performStartTime);
     return result;
   }
 
-  private void emitTimerMetric(final String metric, final long time)
+  private void emitTimerMetric(final String metric, final TaskAction<?> action, final long time)
   {
     final ServiceMetricEvent.Builder metricBuilder = ServiceMetricEvent.builder();
     IndexTaskUtils.setTaskDimensions(metricBuilder, task);
+    final String actionType = getActionType(toolbox.getJsonMapper(), action);
+    if (actionType != null) {
+      metricBuilder.setDimension("taskActionType", actionType);
```

Comment:
Side question: Is there a difference between emitting a `null` value for a dimension and just not emitting it? If not, maybe event builder itself should have this null check.

Reply:
I'm not sure really. I guess the generated JSON might be different, if the event builder doesn't omit null dimensions, but I'm not sure if that has meaningful effect.